pkg/daemon/daemon_test.go

```diff
@@ -470,3 +470,54 @@ func TestDaemon_ProcessSynceLogs(t *testing.T) {
 
 	}
 }
+
+type populateAndRenderPtp4lConfTestCase struct {
+	profileName    string
+	testPtp4lConf  string
+	expectedOutput string
+	testName       string
+	iface          string
+	messageTag     string
+	socketPath     string
+	pProcess       string
+}
+
+func InitPopulateAndRenderPtp4lConfTestCase() []populateAndRenderPtp4lConfTestCase {
+	ret := []populateAndRenderPtp4lConfTestCase{
+		{
+			profileName:    "OC",
+			testPtp4lConf:  "[global]\nslaveOnly 1",
+			expectedOutput: "#profile: OC\n\n[ens2f0]\n[global]\nslaveOnly 1\nmessage_tag test message_tag\nuds_address /var/run/ptp4l.0.socket",
+			testName:       "Basic OC test",
+			iface:          "ens2f0",
+			messageTag:     "test message_tag",
+			socketPath:     "/var/run/ptp4l.0.socket",
+			pProcess:       "",
+		},
+		{
+			profileName:    "BC",
+			testPtp4lConf:  "[global]\nslaveOnly 0\n[ens1f0]\nmasterOnly 0\n[ens1f1]\nmasterOnly 1",
+			expectedOutput: "#profile: BC\n\n[ens1f0]\nmasterOnly 0\n[ens1f1]\nmasterOnly 1\n[global]\nslaveOnly 0\nmessage_tag test message_tag\nuds_address /var/run/ptp4l.0.socket",
```

Comment:
Can we do something like this I think for tests its better if its easier to read. 
```suggestion
			expectedOutput: `
	#profile: BC
	
	[ens1f0]
	masterOnly 0
	[ens1f1]
	masterOnly 1
	[global]
	slaveOnly 0
	message_tag test message_tag
	uds_address /var/run/ptp4l.0.socket`,
```

Reply:
> Can we do something like this I think for tests its better if its easier to read.

That was failing because of mismatched line endings.

Reply:
Probably a leading newline, could use `strings.TrimSpace` on the the expected. As will remove the spaces from the start and the end where you wouldn't expect the rendered output to have any but I'm not that bothered just thought it might make the test cases easer to understand.